src/tvheadend/Subscription.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+/*
+ *      Copyright (C) 2005-2015 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, write to
+ *  the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+ *  http://www.gnu.org/copyleft/gpl.html
+ *
+ */
+
+#include <string>
+#include <mutex>
+#include <condition_variable>
+
+#include "../HTSPTypes.h"
+
+extern "C"
+{
+#include "libhts/htsmsg.h"
+}
+
+class CHTSPConnection;
+
+class Subscription
+{
+public:
+  Subscription(CHTSPConnection &conn);
+
+  bool              IsActive() const;
+  uint32_t          GetId() const;
+  uint32_t          GetChannelId() const;
+  uint32_t          GetWeight() const;
+  int32_t           GetSpeed() const;
+  eSubsriptionState GetState() const;
+
+  /**
+   * Subscribe to a channel on the backend
+   * @param channelId the channel to subscribe to
+   * @param weight the desired subscription weight
+   * @param restart restart the current subscription (i.e. after lost connection), other parameters will be ignored
+   */
+  void SendSubscribe(uint32_t channelId = 0, uint32_t weight = SUBSCRIPTION_WEIGHT_NORMAL, bool restart = false);
```

Comment:
Because we call "m_subscription.SendSubscribe(true)" when the connection to the backend is restored, we do not care what the channelId is in this case as we want just to restart the current stream (with the curret channel id)